### DISCUSSION TITLE
Amazing.Theme.NiceAdmin Theme

### DISCUSSION THREAD

**Discussion by vnetonline** [oqtane/oqtane.framework#2915] [Show and tell]
I am going to have a go at building an Amazing.Theme.NiceAdmin now that the flashing issue has been resolved by reverting to the old way #2850 

I have opened this discussion so that I can log the progress as I build the theme and ask for help when I get stuck.

The goal is to have something which looks like this 

![image](https://github.com/oqtane/oqtane.framework/assets/15242923/55f021d6-50e5-44fc-9f69-788db3f02ea7)

**Comment by vnetonline** [comment]
@sbwalker & @leigh-pointer I have made good progress with the Amazing.Theme.NiceAdmin, however the custom controls I make are in the namespace Amazing.Theme.NiceAdmin.Controls and the solution structure look like below ..... is it the right directory to put the controls 

![image](https://github.com/oqtane/oqtane.framework/assets/15242923/e7f5d081-5862-42ce-8cb4-e87e59c01088)

When I call @ThemePath() it picks up the namespace Amazing.Theme.NiceAdmin.Controls but all my resources are in namespace Amazing.Theme.NiceAdmin should I make my own LoginBase?

**Comment by sbwalker** [comment]
@vnetonline if you want to create controls that are specific to your custom theme, I would suggest that you put them in a /controls folder but use the same namespace as your theme (ie. @namespace YourTheme). You will need to name the component files in a unique way so they don't conflict with the theme or container files which are also part of the theme (ie. MenuControl.razor). If you use this approach the ThemePath will be consistent for all components within the theme.

**Comment by vnetonline** [comment]
@sbwalker which /controls folder

**Reply by sbwalker** [reply]
If these are controls for your custom theme then just create a /controls folder in the Client project for your theme. Actual folder structure does not matter - it's the namespace which is most important to Oqtane.

**Comment by vnetonline** [comment]
Just thought I would give you a preview of something amazing coming,

Amazing NiceAdmin Theme Tasks

- [x] Main Default Theme
- [x] Default Card Container
- [x] Menu System
- [x] Login and Logout 
- [ ] Notifications
- [ ] User Profile
- [ ] Components
    - [ ] Alerts
    - [ ] Accordian
    - [ ] Badges
    - [ ] Breacrumbs
    - [ ] Buttons
    - [ ] Cards
        - [ ] Default Card
    - [ ] Carousel
    - [ ] List Group
    - [ ] Modal
    - [ ] Tabs
    - [ ] Pagination
    - [ ] Progress
    - [ ] Spinners
    - [ ] ToolTips 
- [ ] Register Page Theme
- [ ] Login Page Theme
- [ ] Contact Page Theme
- [ ] F.A.Q Page Theme
- [ ] Error Pages Theme (404, 500)
- [ ] Dashboard Theme

![screen2gif-nicetheme-demo1](https://github.com/oqtane/oqtane.framework/assets/15242923/4606df43-9647-4818-afe1-d1f4fe79894b)

**Comment by vnetonline** [comment]
While building the `Notifications` component raised an issue #2974 and submitted a Pull request  #2975 to introduce the ability to have an `IsRead` property in `Notifications`

**Comment by vnetonline** [comment]
When working on a theme there is no ability to nicely handle Theme based errors raised an issue #2976 to get @sbwalker and @leigh-pointer thoughts.

**Reply by leigh-pointer** [reply]
I have handled this internal but i am all for something standardized!

**Comment by vnetonline** [comment]
Notifications Feature Complete just waiting for #2975 to be merged as it has a dependency on the pull request

![image](https://github.com/oqtane/oqtane.framework/assets/15242923/59e88e73-583e-499f-bbfe-7fbf96453621)

**Reply by lanthonyneville** [reply]
Any update on this?

**Reply by vnetonline** [reply]
Working on it

**Reply by SunSecond** [reply]
If [sbwalker](https://github.com/sbwalker) perfectly integrate this theme into oqtane, replace the old Admin part, and completely separate the user end and management end of oqtane, it will make Oqtane more powerful and flexible.

**Reply by sbwalker** [reply]
@SunSecond this theme is being developed by @vnetonline and is not intended to become the default theme in the framework. However I do expect that @vnetonline will make it available as either an open source or commercial extension.

**Comment by oqtanecms** [comment]
@vnetonline where to test this cool theme ?

**Reply by vnetonline** [reply]
I am working on the theme it will be release soon after https://learnoqtane.com is launched
